src/schnorr.rs

```diff
@@ -1,31 +1,57 @@
-use super::Barretenberg;
+use super::{Barretenberg, Error};
 
 pub(crate) trait SchnorrSig {
-    fn construct_signature(&self, message: &[u8], private_key: [u8; 32]) -> [u8; 64];
-    fn construct_public_key(&self, private_key: [u8; 32]) -> [u8; 64];
-    fn verify_signature(&self, pub_key: [u8; 64], sig: [u8; 64], message: &[u8]) -> bool;
+    fn construct_signature(&self, message: &[u8], private_key: [u8; 32])
+        -> Result<[u8; 64], Error>;
+    fn construct_public_key(&self, private_key: [u8; 32]) -> Result<[u8; 64], Error>;
+    fn verify_signature(
+        &self,
+        pub_key: [u8; 64],
+        sig: [u8; 64],
+        message: &[u8],
+    ) -> Result<bool, Error>;
 }
 
 #[cfg(feature = "native")]
 impl SchnorrSig for Barretenberg {
-    fn construct_signature(&self, message: &[u8], private_key: [u8; 32]) -> [u8; 64] {
+    fn construct_signature(
+        &self,
+        message: &[u8],
+        private_key: [u8; 32],
+    ) -> Result<[u8; 64], Error> {
         let (sig_s, sig_e) = barretenberg_sys::schnorr::construct_signature(message, private_key);
 
-        let sig_bytes: [u8; 64] = [sig_s, sig_e].concat().try_into().unwrap();
-        sig_bytes
+        let sig_bytes: [u8; 64] = [sig_s, sig_e]
+            .concat()
+            .try_into()
+            .map_err(Error::SchnorrConvert)?;
```

Comment:
I'm confused. If this were handled by the type system, we wouldn't need to `try_into` and `unwrap`? Is there some other way to concat these that doesn't require the `try_into`?

Note that even the `some_usize_number as i32` syntax we use a lot isn't actually safe and will panic on some architectures. We should likely be using `try_into` on those too...

Reply:
Well, it's _almost_ enforced. I guess what I mean is that correctness of this code is determined at compile time so we should try to avoid adding runtime checks on this.

We know that `construct_signature` will always return two 32 byte arrays and so we can concat them into a 64 byte array and that will work no matter what the inputs to the function are. We can make this explicit by adding a type declaration for the return values from `construct_signature` to be safer though.

Reply:
I've made a PR https://github.com/noir-lang/aztec_backend/pull/163 which makes these conversions safer by centralising them into a single function which can be easily reasoned about.